app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,44 @@
+class TransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_category
+
```

Comment:
Hello @joel-kalema , I appreciate your suggestion, and you're right, it would be a better approach. However, the current project requirements don't mention it explicitly. I'll take your advice into consideration and plan to implement it over the weekend. Thank you for your input.